src/data/sym2.lean

```diff
@@ -94,6 +94,21 @@ begin
   { cases h; rw [h.1, h.2], rw eq_swap }
 end
 
+/-- The universal property of `sym2`; symmetric functions of two arguments are equivalent to
+functions from `sym2`. Note that this applies to both `Prop` and `Type`, although frequently
+`sym2.from_rel` is a more convenient spelling when working with `Prop`. -/
+def lift {β : Sort*} :
+  {f : α → α → β // ∀ a₁ a₂, f a₁ a₂ = f a₂ a₁} ≃ (sym2 α → β) :=
+{ to_fun := λ f, quotient.lift (λ a : α × α, (f : α → α → β) a.1 a.2) $
```

Comment:
Could you please make that `to_fun` a separate definition? We needed that generalisation of `sym2.from_rel` some time ago.

Reply:
What would be the point of the separate definition? If you use `lift` it unfolds to this anyway.

Reply:
Well I guess it's easier to have a definition without mentioning a subtype.
```
def from_fun {β : Sort*} {f : α → α → β} (comm : ∀ ⦃x y⦄, f x y = f y x) :
  sym2 α → β :=
quotient.lift (uncurry f) (by { rintro ⟨x, y⟩ _ ⟨_, _⟩, {refl }, unfold uncurry, rw comm })
```

Reply:
I think that def results in ` {β : Type*}` since curry isn't defined on sorts

Reply:
Oh that's annoying... There's no real reason for it, right?

Reply:
Wait, what? The original definition of `from_rel` uses `function.uncurry r`. What is this witchcraft?

Reply:
Wait, what? The original definition of `from_rel` uses `function.uncurry r`. What is this witchcraft?

Reply:
Looking again, it looks like there's no point in letting `lift` work on `Sort` anyway, no one every needs a subtype of a proof, and `quotient.induction` works just fine for the cases when Sort would apply.